### ISSUE TITLE
package dependency

### ISSUE THREAD

**Issue by aremazeilles** [tecnalia-advancedmanufacturing-robotics/ros_node_test#7]
ping @miguelprada 

I am bringing this functionality to a [ros_comm fork](https://github.com/aremazeilles/ros_comm/tree/service_filter_check).

The question is related to how properly define the dependency on [rospy_message_converter](http://wiki.ros.org/rospy_message_converter).

In this repo, I have defined it this way:

In [package.xml](https://github.com/tecnalia-advancedmanufacturing-robotics/ros_node_test/blob/master/node_test/package.xml#L15):

```xml
  <depend>rospy_message_converter</depend>
```

In [CMakeLists.txt](https://github.com/tecnalia-advancedmanufacturing-robotics/ros_node_test/blob/master/node_test/CMakeLists.txt#L6)

```make
find_package(catkin REQUIRED COMPONENTS
    rospy
    rospy_message_converter
)
```

and further on:

```make
catkin_package(
    CATKIN_DEPENDS
        rospy
        rospy_message_converter
)
```

If I look to the upstream ros_comm config in the [rostest package](https://github.com/ros/ros_comm/tree/melodic-devel/tools/rostest), I find:

* in [package.xml](https://github.com/ros/ros_comm/blob/melodic-devel/tools/rostest/package.xml), most of the dependencies set as `run_depend`.
* in [CMakeLists.txt](https://github.com/ros/ros_comm/blob/melodic-devel/tools/rostest/CMakeLists.txt), there is very little package dependency placed at the level of `find_package` and of `catkin_package`.

Is it because, being a python package, it is sufficient to state the dependency as a run dependency, and so there is no need to place it as dependency in the CMakeLists.txt, as it is not needed for compiling / installing purposes?

I am a bit lost here. Would you have a better understanding of this? 

Thanks

**Comment by miguelprada** [comment]
> Is it because, being a python package, it is sufficient to state the dependency as a run dependency, and so there is no need to place it as dependency in the CMakeLists.txt, as it is not needed for compiling / installing purposes?

I think you're right:

- `find_package` is used to _import_ libraries into your CMake script (e.g. defines variables such as `${some_package_INCLUDE_DIRS}`, or exported targets in modern CMake), so you can use them to build your C++ targets. Adding libraries to the `find_package(catkin COMPONENTS ...)` instead of using separate `find_pacakge` calls is just a convenience tool, that mainly aggregates everything into the `catkin_INCLUDE_DIRS` and `catkin_LIBRARIES`.
- `catkin_package` is, in turn, used to facilitate exporting the proper flags from your package to dependent C++ targets, so they can build against anything you may export from your package (most commonly headers and binary libraries).

With `rospy_message_converter` being a pure Python library, there's really no point in `find_package`-ing it that I'm aware of. Likewise, if everything you _export_ from your package is Python modules/scripts, I think it's safe to issue an empty `catkin_package()`.

I think this the above applies to `rospy` as well.

**TL;DR;** I'm pretty sure you only need to add `rospy` and `rospy_message_converter` as `exec_depend` to your package (since you're using manifest format 2), and that you can replace [all of this](https://github.com/tecnalia-advancedmanufacturing-robotics/ros_node_test/blob/3e8624ab44041130e059829b8bef16eb828c1afe/node_test/CMakeLists.txt#L4-L40) with

```cmake
find_package(catkin REQUIRED)
catkin_python_setup()
catkin_package()
```
